crates/blockchain-tree/src/block_buffer.rs

```diff
@@ -105,7 +105,8 @@ impl BlockBuffer {
 
     /// Discard all blocks that precede block number from the buffer.
     pub fn remove_old_blocks(&mut self, block_number: BlockNumber) {
-        let mut block_hashes_to_remove = Vec::new();
+        let blocks_to_remove = self.earliest_blocks.range(..=block_number).count();
+        let mut block_hashes_to_remove = Vec::with_capacity(blocks_to_remove + 1);
```

Comment:
i'd be curious to see the benchmark numbers as i think this should actually be slower in some situations, given we're iterating twice? once to get the count, once to add the blocks

Reply:
actually quite similar for 100, and a bit better for 1000 and for 10000 yeah significantly faster

Reply:
can you provide the numbers before and after? the `range` iterates a linked list, so i really can't imagine it's faster

Reply:
it's significant for 1000 too, should I push it ?